test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,670 @@
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/test.h"
+#include "test_harness/util/api_const.h"
+#include "test_harness/workload/random_generator.h"
+
+/*
+ * In this test, we want to verify the usage of the cursor bound API and check that the cursor
+ * returns the correct key when bounds are set.
+ * During the test duration:
+ *  - M threads will keep inserting new random keys.
+ *  - N threads will execute search_near calls with random bounds set. Each search_near
+ * call with bounds set is verified using the standard cursor's search and next/prev calls.
+ *  - O threads will continuously remove random keys.
+ *  - P threads will continuously update random keys.
+ *  - Q threads will utilize the custom operation and will execute next() and prev() calls with
+ * random bounds set. Both next() and prev() calls with bounds set is verified against the
+ * default cursor next() and prev() calls.
+ */
+namespace test_harness {
+
+class cursor_bound_01 : public test {
+    /* Struct helper to represent the lower and uppers bounds for the range cursor. */
+    struct bound {
+        std::string key;
+        bool inclusive;
+        bool lower_bound;
+
+        bound() = default;
+        bound(uint64_t key_size_max, bool _lower_bound) : lower_bound(_lower_bound)
+        {
+            bool set_inclusive = random_generator::instance().generate_integer(0, 1);
+            auto key_size =
+              random_generator::instance().generate_integer(static_cast<uint64_t>(1), key_size_max);
+            auto random_key = random_generator::instance().generate_random_string(
+              key_size, characters_type::ALPHABET);
+            key = random_key;
+            inclusive = set_inclusive;
+        }
+
+        std::string
+        get_config()
+        {
+            return "bound=" + std::string(lower_bound ? "lower" : "upper") +
+              ",inclusive=" + std::string(inclusive ? "true" : "false");
+        }
+    };
+
+    const uint64_t MAX_ROLLBACKS = 100;
+    enum bounds { NO_BOUNDS, LOWER_BOUND_SET, UPPER_BOUND_SET, ALL_BOUNDS_SET };
+    bool reverse_collator_enabled = false;
```

Comment:
I believe this is technically a private member, can we get an _ in front of the variable name. The MAX_ROLLBACKS and enum are a bit unusual so I don't think we have naming standards for them.

Reply:
I think we don't need to change MAX_ROLLBACKS or the enum.